core/client/fs/src/main/java/alluxio/client/block/StreamFactory.java

```diff
@@ -75,55 +67,26 @@ public static BlockOutStream createRemoteBlockOutStream(FileSystemContext contex
    * @param blockId the block ID
    * @param blockSize the block size in bytes
    * @param address the Alluxio worker address
+   * @param openUfsBlockOptions the options to open a UFS block, set to null if this is block is
+   *        not persisted in UFS
    * @param options the in stream options
    * @return the {@link InputStream} object
    */
-  public static BlockInStream createLocalBlockInStream(FileSystemContext context, long blockId,
-      long blockSize, WorkerNetAddress address, InStreamOptions options) {
-    if (NettyUtils.isDomainSocketSupported(address)) {
-      return BlockInStream
-          .createNettyBlockInStream(blockId, blockSize, address, context, options);
+  public static BlockInStream createBlockInStream(FileSystemContext context, long blockId,
+      long blockSize, WorkerNetAddress address, Protocol.OpenUfsBlockOptions openUfsBlockOptions,
+      InStreamOptions options) {
+    if (address.getHost().equals(NetworkAddressUtils.getClientHostName()) && Configuration
+        .getBoolean(PropertyKey.USER_SHORT_CIRCUIT_ENABLED) && !NettyUtils
+        .isDomainSocketSupported(address)) {
+      try {
+        return BlockInStream
+            .createShortCircuitBlockInStream(blockId, blockSize, address, context, options);
+      } catch (NotFoundException e) {
+        // Ignore.
```

Comment:
please javadoc here why we could ignore.
Also, should we ignore exceptions for `createBlockOutStream` similarly?